src/pybind/mgr/cephadm/module.py

```diff
@@ -1231,6 +1214,47 @@ def _run_cephadm(self,
     def _get_hosts(self, label: Optional[str] = '', as_hostspec: bool = False) -> List:
         return list(self.inventory.filter_by_label(label=label, as_hostspec=as_hostspec))
 
+    def _copy_cephadm(self, host: str, addr: str):
+        """
+        Copy the cephadm script provided in the cephadm manager module
+        """
+        # Refresh the copy in memory
+        self._read_cephadm_script()
+
+        # Copy the script to the host
+        with self._remote_connection(host, addr) as tpl:
+            conn, connr = tpl
+            try:
+                cephadm_path = self.get_ceph_option('cephadm_path')
+                out, err, code = remoto.process.check(
+                    conn,
+                    ['mkdir', '-p', os.path.dirname(cephadm_path)])
+                self.log.debug('mkdir-> %s-:-%s' % (out, err))
+                out, err, code = remoto.process.check(
+                    conn,
+                    ['cp', '/dev/stdin' , cephadm_path],
+                    stdin=self._cephadm.encode('utf-8'))
```

Comment:
prone to races. Might make sense to do *one* of those:

1. use a unique name
2. copy to a temp file name
3. execute a `mv` to the final file name